base_iso3166/models/res_country.py

```diff
@@ -27,18 +27,18 @@ class ResCountry(models.Model):
 
     @api.depends("code")
     def _compute_codes(self):
-        for country in self:
+        self.code_alpha3 = False
+        self.code_numeric = False
+
+        for country in self.filtered("code"):
             c = False
-            for country_type in ["countries", "historic_countries"]:
-                try:
-                    c = getattr(pycountry, country_type).get(alpha_2=country.code)
-                except KeyError:
-                    c = getattr(pycountry, country_type).get(alpha2=country.code)
-                if c:
-                    break
+            if country.code:
```

Comment:
Remove this `if` and restore the previous code, and you still have several commits.